### ISSUE TITLE
Prepare release-1.3 branches

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#16201]
This issue aims to track all steps needed to prepare the release 1.3 branches/builds.

Please add all steps taken here so we can document for future releases

- [ ] (Optional) Update go dependencies. This must be done **before branching** so that PRs can be easily backported later. `go get ... && go get ... && go mod tidy` (then on istio/istio only: `go mod vendor`)
  - [ ] istio/istio
  - [ ] istio/pkg
  - [ ] istio/api
  - [ ] istio/cni
  - [ ] istio/operator
  - [ ] istio/tests
  - [ ] istio/client-go
- [x] Create a `release-1.3` branch in every Istio repo - run commands `export org=xyz ; export repo=xyz ; (git clone git@github.com:${org}/${repo}.git && cd $repo && git checkout -b release-1.3 && git push --set-upstream origin release-1.3)`. (owner: @rlenglet)
  - [x] [istio/istio](https://github.com/istio/istio/tree/release-1.3)
  - [x] [istio/pkg](https://github.com/istio/pkg/tree/release-1.3)
  - [x] [istio/api](https://github.com/istio/api/tree/release-1.3)
  - [x] [istio/cni](https://github.com/istio/cni/tree/release-1.3)
  - [x] [istio/proxy](https://github.com/istio/proxy/tree/release-1.3)
  - [x] [istio/envoy](https://github.com/istio/envoy/tree/release-1.3) (branch from [envoyproxy/envoy-wasm:master](https://github.com/envoyproxy/envoy-wasm/tree/master) not from [istio/envoy:master](https://github.com/istio/envoy/tree/master), cf. @PiotrSikora)
  - [x] [istio/operator](https://github.com/istio/operator/tree/release-1.3)
  - [x] [istio/installer](https://github.com/istio/installer/tree/release-1.3)
  - [x] [istio-releases/pipeline](https://github.com/istio-releases/pipeline/tree/release-1.3)
  - [x] istio/tests  - not needed for 1.3 since this is a new repo
  - [x] istio/client-go  - not needed for 1.3 since this is a new repo
- [x] Commit a PR into every `release-1.3` branch (except istio-releases/pipeline) to modify every `CODEOWNERS` to contain only the release managers for this release. (owner: @rlenglet) Command: `export org=xyz ; export repo=xyz ; (git clone git@github.com:${org}/${repo}.git && cd $repo && git checkout release-1.3 && git checkout -b release-1.3-codeowners && echo '* @istio/release-managers-1-3' > CODEOWNERS && git add CODEOWNERS && git commit -m 'Set release managers as CODEOWNERS for release-1.3' && git push --set-upstream origin release-1.3-codeowners)`. Those PRs need to be force-merged using admin privileges.
  - [x] https://github.com/istio/istio/pull/16246
  - [x] https://github.com/istio/pkg/pull/54
  - [x] https://github.com/istio/api/pull/1035
  - [x] https://github.com/istio/cni/pull/140
  - [x] https://github.com/istio/proxy/pull/2356
  - [x] https://github.com/istio/envoy/pull/89 (has no effect since `CODEOWNERS` is not enabled in that repo)
  - [x] https://github.com/istio/operator/pull/189
  - [x] https://github.com/istio/installer/pull/324
  - [x] istio/tests - not needed for 1.3 since this is a new repo
  - [x] istio/client-go - not needed for 1.3 since this is a new repo
- [x] Disable Prow's `merges-blocked-needs-admin` check for the `release-1.3` branch. (owner: @howardjohn) https://github.com/istio/test-infra/pull/1662
- [x] Set up prow on release branches (owner: @howardjohn)
  - [x] istio/istio https://github.com/istio/test-infra/pull/1648
  - [x] istio/api https://github.com/istio/test-infra/pull/1659
  - [x] istio/pkg - uses same config for master and release - no changes needed. This is ok as long as we don't need to change the prow config (which is unlikely to change)
  - [x] istio/cni https://github.com/istio/test-infra/pull/1654
  - [x] istio/proxy https://github.com/istio/test-infra/pull/1653
  - [x] istio/operator https://github.com/istio/test-infra/pull/1655
  - [x] istio/installer - not yet on prow
  - [x] istio-releases/pipelines https://github.com/istio/test-infra/pull/1656
  - [x] istio/tests - not needed for 1.3 since this is a new repo
  - [x] istio/client-go - not needed for 1.3 since this is a new repo
- [x] Set up branch protection (owner: @howardjohn) https://github.com/istio/test-infra/pull/1649
- [x] Update all repos that point to master branch of another repo to point to the release branch
  - [x] istio/istio https://github.com/istio/istio/pull/16479
  - [x] istio/proxy https://github.com/istio/proxy/pull/2372
  - [x] istio/installer https://github.com/istio/installer/pull/345
  - [x] istio/operator https://github.com/istio/operator/pull/216
  - [x] istio/cni https://github.com/istio/cni/pull/150
  - [x] istio/tests - not needed for 1.3 since this is a new repo
  - [x] istio/client-go - not needed for 1.3 since this is a new repo
- [x] ~~Point `go.mod` to release branches (owner: TBD) `go get istio.io/istio/xyz:sha` for each dependency (where `sha` is the hash of the commit at the HEAD of the `release-1.3` branch), then `go mod vendor`. TODO: I think this is not needed -- go modules aren't pointing to a branch unlike dep, so we just need to make sure when we do update it we point to the right release branch.~~
- [x] Setup daily builds
  - [x] setup periodics that trigger daily builds (owner: @howardjohn) https://github.com/istio/test-infra/pull/1657
  - [x] trigger daily build manually and ensure it works https://github.com/istio-releases/pipeline/pull/1027
- [x] Run the long-running tests on the daily builds (owners: @mandarjog, @howardjohn, @JimmyCYJ)
- [x] Send announcement on Discuss (owner: @geeknoid)

Once all this is done, create a first snapshot. https://github.com/istio/istio/issues/16411

**Comment by geeknoid** [comment]
Need istio/pkg as well.

**Comment by howardjohn** [comment]
All prow PRs are created. Some are blocked by the release branch being create, but most can be merged at any time

**Comment by rlenglet** [comment]
Re: this comment in the description:

> TODO: I think this is not needed -- go modules aren't pointing to a branch unlike dep, so we just need to make sure when we do update it we point to the right release branch.

I still think we need to update all the `go.mod` dependencies to point to commits in `release-1.3` branches, because we'll create release tags on those commits and I don't see the point of creating release branches and release tags if we're not using those commits to build the release.

**Comment by sdake** [comment]
necessary merge for 1.3: https://github.com/istio/istio/pull/16131

This PR pins the base image.  The advantages:
1. generated by CI per z-stream release.
1. confined to one product version instead of several as is now the case.
1. solves the compliance problem of CVEs in the base image - by simply generating a new build rather than doing an in-depth analysis of the CVE as it relates to Istio.  This will happen naturally as patch releases are created, rather than never as is currently the case.
1. permits us to use Ubuntu's upstream CVE process for our base, rather than depend solely on our own overworked security team.

**Comment by howardjohn** [comment]
We need to merge either https://github.com/istio/test-infra/pull/1658 or https://github.com/istio/test-infra/pull/1652. I don't care which but one needs to get in to unblock the release-1.3 creation

@geeknoid

**Comment by howardjohn** [comment]
Last task needed is to send an announcement on discuss and/or slack @geeknoid 

and trigger a snapshot build, need an owner

**Comment by brian-avery** [comment]
I'll take care of the announcement.

**Comment by brian-avery** [comment]
The announcement is now posted here: https://discuss.istio.io/t/1-3-release-branch-cut/3484 and in the general slack channel

**Comment by rlenglet** [comment]
`istio/envoy` doesn't seem to have `CODEOWNERS` enabled. Is that intentional?
Cf. https://github.com/istio/envoy/pull/93, a backport into `istio/envoy:release-1.3`, which didn't require approval from @istio/release-managers-1-3  despite being listed in [`CODEOWNERS`](https://github.com/istio/envoy/blob/release-1.3/CODEOWNERS).

**Comment by geeknoid** [comment]
The Istio CI system is completely ignorant of the istio/envoy repo. We're
not enforcing any setting on the repo (such as code owners), not running
any builds, nothing.

There's nothing stopping us if we need this. Just let me know.

**Comment by rlenglet** [comment]
OK, that's fine. It was just surprising.
